core/src/main/java/com/digitalpebble/stormcrawler/protocol/HttpHeaders.java

```diff
@@ -27,7 +27,9 @@
  * @see <a href="http://rfc-ref.org/RFC-TEXTS/2616/">Hypertext Transfer Protocol -- HTTP/1.1 (RFC
  *     2616)</a>
  */
-public interface HttpHeaders {
+public final class HttpHeaders {
```

Comment:
What's wrong with having an interface for this?

Reply:
`HttpHeaders` is a utility class, meaning that it only has static, stateless functions without any other dependencies. But if you use an interface or abstract class for a utility class you communicate to the framework user that implementing this interface or inheriting from this abstract class is supposed to happen because the interface/abstract class is needed somewhere.

When you look at the java standard library they declare utility-classes like I do. (Except for the `final` declaration, thats just my personal taste to make it 100% clear, that noone can ever use this class for something else than it's static functions.)

```java
package java.util;
....

/**
 * This class contains various methods for manipulating arrays (such as
 * sorting and searching). This class also contains a static factory
 * that allows arrays to be viewed as lists.
 ...
 * @since  1.2
 */
public class Arrays {

    /**
    ...
     */
    private static final int MIN_ARRAY_SORT_GRAN = 1 << 13;

    // Suppresses default constructor, ensuring non-instantiability.
    private Arrays() {}

    ...
```

There are some use-cases for static methods in interfaces (see `java.util.stream.Stream`) but as long as an interface/abstract class doesn't declare any non-static methods and/or the static functions do not take/return the containing interface/abstract class we should implement utility classes like shown in the standard library for a clear intention communication.

Reply:
OK, my perception of it probably dates back to when doing this as an interface was in fashion (whenever that was)